File: exercicios/python/Exercicio05/aluno.py
from usuario_bu import UsuarioBU
from abc import ABC, abstractmethod

class Aluno(UsuarioBU):

    @abstractmethod
    def __init__ (self, cpf:int, dias_de_emprestimo:int, matricula:int):
        super().__init__(cpf, dias_de_emprestimo)
        self.__matricula = matricula

    @property
    def matricula(self):
        return self.__matricula
    
    @matricula.setter
    def matricula(self, matricula):
        self.__matricula = matricula
    
    def emprestar (self, titulo_do_livro:str):
        return 'de matricula "' + str(self.__matricula) + super().emprestar(titulo_do_livro)

    def devolver (self, titulo_do_livro:str):
        return 'Aluno de matricula "' + str(self.__matricula) + super().devolver(titulo_do_livro)

File: exercicios/java/Exercicio05/src/main/java/org/example/Aluno.java
package org.example;

public abstract class Aluno extends UsuarioBU {
    private int matricula;

    public Aluno(int matricula, long cpf, int diasDeEmprestimo) {
        super(cpf, diasDeEmprestimo);
        this.matricula = matricula;
    }
    public int getMatricula() {
        return matricula;
    }
    public void setMatricula(int matricula) {
        this.matricula = matricula;
    }

    @Override
    public String devolver(String tituloLivro) {
        String emprestarDoUsuario = super.devolver(tituloLivro);
        String strMatricula = String.valueOf(this.matricula);
        return "Aluno de matricula \"" + strMatricula + emprestarDoUsuario;
    }
    @Override
    public String emprestar(String tituloLivro) {
        String emprestarDoUsuario = super.emprestar(tituloLivro);
        String strmatricula = String.valueOf(this.matricula);
        return "de matricula \"" + strmatricula + emprestarDoUsuario;
    }
}
